### PR TITLE
converted coc to rst and put links in code_of_conduct.md

### PR DESCRIPTION
**Pull request by story645**
Converted the CoC so that it could be integrated into our docs - it's current form as an md file makes it feel a bit out of place. Not totally sure this will work but tried using the git redirect template inside code_of_conduct.md to keep the coc link working. Any ideas to test would be great.

### CODE REVIEW

**Comment by QuLogic** [comment]
Unfortunately, I think we need this file for [GitHub's purposes](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project), unless it understands the redirect you've added.

File: doc/users/project/code_of_conduct.rst [hunk 1]
```diff
@@ -0,0 +1,147 @@
+.. code_of_conduct
+
+====================================
+Contributor Covenant Code of Conduct
```
Comment (jklymak) [inline doc/users/project/code_of_conduct.rst:4]: Orthogonal to this PR, but came up because you needed a modified entry in the TOC:

Is this the "Contributor Covenant" code of conduct, or is it " adapted from the `Contributor Covenant` "? 

This headline is missing punctuation or something because "Contributor Covenant Code of Conduct" doesn't mean anything in plain English, and only makes sense if you know what "Contributor Covenant" is, which we don't explain until the bottom.  

On-topic: I don't see what is wrong with moving this, but maybe some people don't want to click through?

**Comment by ksunden** [comment]
My instinct is to say that this is a doc-only change that does not impact the release artifacts, we have the ability to update the docs decoupled from the release itself, and thus not to hold up releasing on this. That said, I can wait until the meeting today to tag, and that may mean we get it in anyway.

**Review by ksunden — APPROVED** [review]
